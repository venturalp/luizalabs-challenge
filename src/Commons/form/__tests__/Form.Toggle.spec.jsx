import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { ThemeWrapper } from 'Commons/tests/Tests.Helpers'
import React from 'react'
import { Toggle } from '../Form.Toggle'

describe('test form toggle', () => {
  it('test toggle uncontrolled', () => {
    const { asFragment } = render(
      <ThemeWrapper>
        <Toggle defaultValue={false} id="toggle" />
      </ThemeWrapper>,
    )

    expect(asFragment()).toMatchSnapshot()

    userEvent.click(screen.getByTestId('toggle'))

    expect(asFragment()).toMatchSnapshot()
  })
})
