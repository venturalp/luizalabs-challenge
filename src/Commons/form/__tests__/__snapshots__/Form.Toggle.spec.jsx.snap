// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test form toggle test toggle uncontrolled 1`] = `
<DocumentFragment>
  <span
    class="sc-bdfBwQ dnfcXD"
    data-testid="toggle"
  >
    <img
      alt="toggleIco"
      src="heartEmpty.png"
    />
  </span>
</DocumentFragment>
`;

exports[`test form toggle test toggle uncontrolled 2`] = `
<DocumentFragment>
  <span
    class="sc-bdfBwQ dnfcXD"
    data-testid="toggle"
  >
    <img
      alt="toggleIco"
      src="heartFilled.png"
    />
  </span>
</DocumentFragment>
`;
